sql/connect/server/src/test/scala/org/apache/spark/sql/connect/pipelines/SparkDeclarativePipelinesServerSuite.scala

```diff
@@ -486,4 +486,138 @@ class SparkDeclarativePipelinesServerSuite
       assert(graphsAfter.isEmpty, "Graph should be removed after drop")
     }
   }
+
+  gridTest("DefineDataset returns resolved data name for default catalog/schema")(
+    Seq(
+      ("TEMPORARY_VIEW", DatasetType.TEMPORARY_VIEW, "tv"),
+      ("TABLE", DatasetType.TABLE, "tb"),
+      ("MV", DatasetType.MATERIALIZED_VIEW, "mv")
+    )
+  ) { case (_, datasetType, datasetName) =>
+    withRawBlockingStub { implicit stub =>
+      val graphId = createDataflowGraph
+      assert(Option(graphId).isDefined)
+
+      val defineDataset = DefineDataset
+        .newBuilder()
+        .setDataflowGraphId(graphId)
+        .setDatasetName(datasetName)
+        .setDatasetType(datasetType)
+
+      val pipelineCmd = PipelineCommand.newBuilder()
+        .setDefineDataset(defineDataset)
+        .build()
+
+      val res = sendPlan(buildPlanFromPipelineCommand(pipelineCmd)).getPipelineCommandResult
+      assert(res !== PipelineCommandResult.getDefaultInstance)
+      assert(res.hasDefineDatasetResult)
+
+      val graphResult = res.getDefineDatasetResult
+      if (datasetName.contains("tv")) {
+        assert(graphResult.getResolvedDataName == s"`${datasetName}`")
+      } else {
+        assert(graphResult.getResolvedDataName == s"`spark_catalog`.`default`.`${datasetName}`")
+      }
+    }
+  }
+
+  gridTest("DefineDataset returns resolved data name for custom catalog/schema")(
+    Seq(
+      ("TEMPORARY_VIEW", DatasetType.TEMPORARY_VIEW,
+        "tv", "custom_catalog", "custom_db", "`tv`"),
+      ("TABLE", DatasetType.TABLE,
+        "tb", "my_catalog", "my_db", "`my_catalog`.`my_db`.`tb`"),
+      ("MV", DatasetType.MATERIALIZED_VIEW,
+        "mv", "another_catalog", "another_db", "`another_catalog`.`another_db`.`mv`")
```

Comment:
nit for readability, we can do something like this: https://github.com/databricks-eng/universe/blob/959a1951d1fef93ee38cd8b35142cb86697b7e72/spark/pipelines/execution/core/test/SQLPipelineSuite.scala#L480-L529